blocks/gu_spdetails/block_gu_spdetails.php

```diff
@@ -317,12 +317,12 @@ public static function return_dates($modname, $activity) {
             case 'forum':
                 $date->gradingduedate = (!empty($activity->cutoffdate)) ?
                                         $activity->cutoffdate :
-                                        (!(empty($date->duedate)) ? $date->duedate + (86400 * 14) : '0');
+                                        (!(empty($date->duedate)) ? $date->duedate : '0');
                 break;
             case 'workshop':
                 $date->gradingduedate = (!empty($activity->gradingduedate)) ?
                                         $activity->gradingduedate :
-                                        (!(empty($date->duedate)) ? $date->duedate + (86400 * 14) : '0');
+                                        (!(empty($date->duedate)) ? $date->duedate : '0');
```

Comment:
Similar to assignment, the `$date->gradingduedate` in workshop should only be based on `$activity->gradingduedate`. If this is empty, just set it to 0.

`$date->gradingduedate = (!empty($activity->gradingduedate)) ? $activity->gradingduedate : '0';`

Reply:
Fixed at commit [1b50080](https://github.com/uofg-andf/andfmoodle/pull/184/commits/1b50080a3cab7bf33148cba444d1b1d2f7579d88)